core/src/Helpers/ExtensionMethods.cs

```diff
@@ -157,14 +157,14 @@ internal static string AsSingleString(this IEnumerable<string> input)
             return String.Join(" ", input);
         }
 
-        internal static SortedSet<string> AsSet(this string singleString)
+        internal static SortedSet<string> AsLowerCaseSortedSet(this string singleString)
```

Comment:
this is not about comparison, this is about transforming date to expected format

Reply:
I'm not sure what you expect here Bogdan. I suppose if this string was Chinese it would just not do anything. (to my knowledge it would not ever be Chinese). Do you suggest to add a comparer? Or do you suggest another way to actually lower case the string before converting it into a set?

---
In reply to: [199543859](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1108#discussion_r199543859) [](ancestors = 199543859)

Reply:
I'm saying that using toLower() is a Java construct to perform case insensitive operations. In .net you initialize your Set with a string comparison and that's it. This has the following advantages: 

- it's fast (every call to ToLower() creates a new string because strings are immutable, whereas with the comparer there is no string duplication)
- it's concise (just initialize your structure with the comparer and that's it)
- it's safe (.net guidelines specify that you should always specify a string comparer explicity ... in my experience it's worth doing this - it will save a few ICMs with customers in Japan :) )

Reply:
Thanks for the clarification on the IComparer 
In this case one new string will be made from which the set is created, however not multiple new strings, right?
The code is generically not multi language enabled and I believe the scenario here may not be subject to globalization challenges. IMO: something also worthwhile taking to our general discussion on code style.